libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2154,7 +2154,10 @@ void TypeChecker::typeCheckABIEncodeCallFunction(FunctionCall const& _functionCa
 		return;
 	}
 
-	solAssert(!functionPointerType->takesArbitraryParameters(), "Function must have fixed parameters.");
+	solAssert(functionPointerType->kind() != FunctionType::Kind::DelegateCall);
+	auto const externallyCallableFunction = functionPointerType->asExternallyCallableFunction(false);
```

Comment:
I'd call it `externalFunctionType` since it's still a function type.

Also, we generally avoid `auto` unless the type is obvious from context - as it was at the point of declaration of `functionPointerType` due to `dynamic_cast`. Here it's not. And when we use auto, we at least include the `*` or `&` to make it clear that it's a pointer/reference.

```suggestion
	FunctionType const* externalFunctionType = functionPointerType->asExternallyCallableFunction(false);
```